pkg/crane/pull.go

```diff
@@ -77,3 +96,73 @@ func pull(_ *cobra.Command, args []string) {
 		log.Fatalf("writing image %q: %v", dst, err)
 	}
 }
+
+func pullLayout(src, dst string) error {
+	// Open the layout.
+	path, err := layout.FromPath(dst)
+	if err != nil {
+		// TODO: This could be cleaner. We can have an entrypoint that inits the layout
```

Comment:
Probably time to add such an entry point. E.g. `FromPathOrNew` could be a helper function encapsulating just the logic here.